### DISCUSSION TITLE
Geometry3D.get_closest_point_to_segment() has inaccurate name

### DISCUSSION THREAD

**Discussion by chardish** [godotengine/godot-proposals#8487] [Scripting]
In the Geometry3D class, the function `get_closest_point_to_segment(point, s1, s2)` has an inaccurate name. Per the documentation, the function

> Returns the 3D point on the 3D segment (s1, s2) that is closest to point. The returned point will always be inside the specified segment.

So this function doesn't get the closest point _to_ a segment. It gets the closest point _on_ the segment to `point`. 

I'd recommend renaming this function `get_closest_point_on_segment`.

(Its sibling function, `get_closest_point_to_segment_uncapped`, has a similar issue and should be renamed `get_closest_point_on_segment_uncapped`.)

**Comment by AThousandShips** [comment]
Note that this can't be done until Godot 5.0 because it breaks compatibility

Also the same naming would apply to the 2D case
